From 1fb4ad8331da9999f2b1b71adf52136683c623c1 Mon Sep 17 00:00:00 2001
From: Corinna Vinschen <xda@vinschen.de>
Date: Tue, 10 Apr 2018 11:57:37 +0200
Subject: [PATCH 6/6] power: fix sysfs_read/sysfs_write usage

* The incoming path to sysfs_read/sysfs_write should be const, as in
  open(2) call.
* Redefine scaling_gov_path as const pointer array.
* Since sysfs_read works on a simple absolute path anyway, make sure
  the scaling_gov_path paths *are* absolute.  Otherwise the code only
  works if Power HAL has / as CWD, which is a bit fragile.

Change-Id: I70c08f8137842569514bcb3f6e0617d46044e6ab
Signed-off-by: Corinna Vinschen <xda@vinschen.de>
---
 utils.c | 14 +++++++-------
 utils.h |  4 ++--
 2 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/utils.c b/utils.c
index 9734086..28798c8 100644
--- a/utils.c
+++ b/utils.c
@@ -47,11 +47,11 @@
 #define USINSEC 1000000L
 #define NSINUS 1000L
 
-char scaling_gov_path[4][80] ={
-    "sys/devices/system/cpu/cpu0/cpufreq/scaling_governor",
-    "sys/devices/system/cpu/cpu1/cpufreq/scaling_governor",
-    "sys/devices/system/cpu/cpu2/cpufreq/scaling_governor",
-    "sys/devices/system/cpu/cpu3/cpufreq/scaling_governor"
+const char *scaling_gov_path[4] ={
+    "/sys/devices/system/cpu/cpu0/cpufreq/scaling_governor",
+    "/sys/devices/system/cpu/cpu1/cpufreq/scaling_governor",
+    "/sys/devices/system/cpu/cpu2/cpufreq/scaling_governor",
+    "/sys/devices/system/cpu/cpu3/cpufreq/scaling_governor"
 };
 
 #define PERF_HAL_PATH "libqti-perfd-client.so"
@@ -127,7 +127,7 @@ static void __attribute__ ((destructor)) cleanup(void)
     }
 }
 
-int sysfs_read(char *path, char *s, int num_bytes)
+int sysfs_read(const char *path, char *s, int num_bytes)
 {
     char buf[80];
     int count;
@@ -155,7 +155,7 @@ int sysfs_read(char *path, char *s, int num_bytes)
     return ret;
 }
 
-int sysfs_write(char *path, char *s)
+int sysfs_write(const char *path, char *s)
 {
     char buf[80];
     int len;
diff --git a/utils.h b/utils.h
index 2080234..3730b39 100644
--- a/utils.h
+++ b/utils.h
@@ -29,8 +29,8 @@
 
 #include <cutils/properties.h>
 
-int sysfs_read(char *path, char *s, int num_bytes);
-int sysfs_write(char *path, char *s);
+int sysfs_read(const char *path, char *s, int num_bytes);
+int sysfs_write(const char *path, char *s);
 int get_scaling_governor(char governor[], int size);
 int get_scaling_governor_check_cores(char governor[], int size,int core_num);
 int is_interactive_governor(char*);
-- 
2.15.1

